common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java

```diff
@@ -436,6 +439,18 @@ public static ClientException clientExceptionFromException(@NonNull final Throwa
 
 
         if (e instanceof OutOfMemoryError) {
+            // Log status code and record exception in span
+            final Span currentSpan = SpanExtension.current();
+
+            // Attach the stack trace, to debug in telemetry later
+            currentSpan.setAttribute(
+                    AttributeName.out_of_memory_exception_stacktrace.name(),
+                    StringUtil.getStacktraceAsStringFromElementArray(e.getStackTrace())
+            );
```

Comment:
given we reach here due to OOM, this (stack trace method does significant allocation) also would likely cause OOM, so I am not sure if we gain anything here.. Still if we do this, this statement should also be under try 
try () {
   span...
} catch (OutOfMemory) {
    /no-op or log
} 

I wonder, If we never had this if block, OOM exception would have bubbled to the top and caused broker process to crash (and perhaps msal side as well) but eventually fixing the issue.

Reply:
This was added in last year July by Melissa, we can check with her if OOM was there for reason or in general to handle UNKNWON_ERRORs. As a fix we can consider moving back earlier state for OOM in march.

Reply:
I think melissa's change was to make unknown errors less common and give us more detailed exceptions, reverting would lead to uptik in unknown errors but functionally the same behavior. I do think a try catch is needed here though, good catch on that

Reply:
I've added the try-catch now